core/server/master/src/main/java/alluxio/master/meta/ServerConfigurationReport.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.wire.ConfigProperty;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is responsible for containing configurations and creating server configuration report.
+ */
+public final class ServerConfigurationReport {
```

Comment:
can we write this class in a general way so that it can be applied to any type of scope? Then we don't need to check isMaster and duplicate logic

Reply:
I create a ConfigRecorder to store lockers, confMap, lostConfMap. 
and init one MasterConfigRecorder and one WorkerConfigRecorder in the ServerConfigurationChecker class. 

I am not quite sure is that the general way? or are there better ways to process the duplicate logic? @aaudiber